app/routes/customer_routes.py

```diff
@@ -0,0 +1,157 @@
+    new_customer = Customer(name=request_body["name"], 
+            postal_code=request_body["postal_code"],
+            phone=request_body["phone"])
+
+    db.session.add(new_customer)
+    db.session.commit()
+    db.session.refresh(new_customer, ["id"])
+
+    return make_response(jsonify({"id":new_customer.id}), 201)
+
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_one_customer(customer_id):
+    customer = validate_model(Customer, customer_id)
+    db.session.delete(customer)
+    db.session.commit()
+    
+    return make_response(jsonify({"id":int(customer_id)}), 200)
+
+
+@customers_bp.route("/<customer_id>", methods=["PUT"])
+def update_one_customer(customer_id):
+    customer = validate_model(Customer, customer_id)
+    request_body = request.get_json()
+    required_attributes = ["name", "phone", "postal_code"]
+
+    validate_request_body(request_body, required_attributes)
+    
+    customer.name = request_body["name"]
+    customer.phone = request_body["phone"]
+    customer.postal_code = request_body["postal_code"]
+    db.session.commit()
+    response_body = request_body
+    response_body['registered_at'] = customer.registered_at
+
+    return make_response(jsonify(response_body), 200)
+
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def get_customer_checked_out_videos(customer_id):
+    validate_model(Customer, customer_id)
+
+    possible_query_params = {"sort" : "", 
+            "count": 0, 
+            "page_num": 0}
+
+    for query_param in possible_query_params:
+        if query_param in request.args:
+            possible_query_params[query_param] = request.args.get(query_param)
+
+    join_query = db.session.query(Rental, Video)\
+            .join(Video, Rental.video_id==Video.id)\
+            .filter(Rental.customer_id == customer_id)
+
+    sort_params = ["title", "release_date"]
+    for param in sort_params:
+        if possible_query_params["sort"] == param:
+            join_query = join_query.order_by(param)
+
+    if possible_query_params["count"] and \
+            possible_query_params["count"].isdigit():
+        possible_query_params["count"] = int(possible_query_params["count"])
+        if possible_query_params["page_num"] and \
+                possible_query_params["page_num"].isdigit():
+            possible_query_params["page_num"] = \
+                    int(possible_query_params["page_num"])
+        else:
+            possible_query_params["page_num"] = 1
+        join_query = join_query.paginate(
+                page=possible_query_params["page_num"], 
+                per_page=possible_query_params["count"]).items
+    else:
+        join_query = join_query.all()
+
+    response_body = []
+    for row in join_query:
+        response_body.append({
+            "id": row.Video.id,
+            "title": row.Video.title,
+            "total_inventory": row.Video.total_inventory,
+            "release_date": row.Video.release_date
+        })
```

Comment:
this was the only way I could access the results of a join query.  I'm not sure how to do it in a different way to make use of the to_dict function.

Reply:
this was the only way I could access the result of the join query. I don't know how to make it in another way to make use of the to_dict function.

Reply:
I agree with Dalia! I tried to access them in the regular way and the attributes of the model object weren't recognized.

Reply:
Ahh, makes sense, because of the join the rows returned by the query are not just Video models. If we only need the video information for the response, we could see if SQLAlchemy has tools for getting the Videos associated with each row, but I think that's far more work than necessary for what we're trying to do here.

Reply:
Just a note when I cloned down the code to test things out there was a migrations folder, but the versions folder had been deleted. In a larger project, we would want to keep the versions in source control so that everyone is sharing the same, most updated version of the database schema.

Reply:
I don't understand why the versions folder isn't there. I can see it in my local version ... and that's the version that we pushed to github. Do you have an idea what we might have done wrong?

Reply:
It looks like your local migrations folders might not have been in sync? I see a couple version files that it looks like were deleted in the last PR on the repo: https://github.com/dnabilali/retro-video-store/pull/11/files, check for files marked as deleted